cmd/sentinel/sentinel.go

```diff
@@ -235,6 +235,10 @@ func (s *Sentinel) GetBestStandby(cv *cluster.ClusterView, keepersState cluster.
 			log.Warningf("ignoring node since its pg state is unknown")
 			continue
 		}
+		if masterState.PGState.SystemID != k.PGState.SystemID {
```

Comment:
But never elect a keeper that we know has a systemID that differs from the current master.